### PR TITLE
Issue#51: Converters functionality

### PR DESCRIPTION
**Pull request by alexorlenko**
dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)

## Code reviewers

- [ ] @reeztl 
- [ ] @Arthurpanda 
- [ ] @mykytam 
- [ ] @omelchenkotimur 

### Second Level Review

- [ ] @kh0ma 
- [ ] @DzigMS 

## Summary of issue

Add mapstruct
Add BaseDto and BaseEntity

## Summary of change

Some change into openapi.yaml

## Testing approach

-

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

**Review by mykytam — APPROVED** [review]

File: home-application/src/main/java/com/softserveinc/ita/homeproject/application/apiservice/NewsApiServiceImpl.java [hunk 1]
```diff
@@ -2,10 +2,14 @@
 
 import com.softserveinc.ita.homeproject.api.NewsApiService;
 import com.softserveinc.ita.homeproject.homeservice.dto.NewsDto;
+import com.softserveinc.ita.homeproject.homeservice.mapperViewToDto.CreateNewsDtoMapper;
```
Comment (kh0ma) [inline home-application/src/main/java/com/softserveinc/ita/homeproject/application/apiservice/NewsApiServiceImpl.java]: `com.softserveinc.ita.homeproject.homeservice.mapper.*`

File: home-application/src/main/java/com/softserveinc/ita/homeproject/application/apiservice/NewsApiServiceImpl.java [hunk 2]
```diff
@@ -15,24 +19,20 @@
...
-        NewsDto newsDto = NewsDto.builder()
-                .title(createNews.getTitle())
-                .description(createNews.getDescription())
-                .text(createNews.getText())
-                .photoUrl(createNews.getPhotoUrl())
-                .source(createNews.getSource())
-                .build();
-        ReadNews response = convertToReadNews(newsService.create(newsDto));
+        NewsDto newsDto = createNewsDtoMapper.convertViewToDto(createNews);
+        NewsDto readNewsDto = newsService.create(newsDto);
```
Comment (kh0ma) [inline home-application/src/main/java/com/softserveinc/ita/homeproject/application/apiservice/NewsApiServiceImpl.java]: readNewsDto -> createdNewsDto

File: home-data/src/main/java/com/softserveinc/ita/homeproject/homedata/entity/BaseEntity.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+import javax.persistence.MappedSuperclass;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+
+@Getter
+@Setter
+@SuperBuilder
+@NoArgsConstructor
+@AllArgsConstructor
+@MappedSuperclass
```
Comment (kh0ma) [inline home-data/src/main/java/com/softserveinc/ita/homeproject/homedata/entity/BaseEntity.java:15]: Eq and Hashcode

File: home-data/src/main/java/com/softserveinc/ita/homeproject/homedata/entity/BaseEntity.java [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+public abstract class BaseEntity {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    protected Long id;
```
Comment (kh0ma) [inline home-data/src/main/java/com/softserveinc/ita/homeproject/homedata/entity/BaseEntity.java]: private

File: home-service/src/main/java/com/softserveinc/ita/homeproject/homeservice/dto/BaseDto.java [hunk 5]
```diff
@@ -0,0 +1,4 @@
+package com.softserveinc.ita.homeproject.homeservice.dto;
+
+public abstract class BaseDto {
```
Comment (kh0ma) [inline home-service/src/main/java/com/softserveinc/ita/homeproject/homeservice/dto/BaseDto.java:3]: id?

File: home-service/src/main/java/com/softserveinc/ita/homeproject/homeservice/dto/NewsDto.java [hunk 6]
```diff
@@ -20,4 +16,64 @@
     private String source;
 
     private String text;
+
+    public NewsDto() {
```
Comment (kh0ma) [inline home-service/src/main/java/com/softserveinc/ita/homeproject/homeservice/dto/NewsDto.java]: Lombok?

File: home-service/src/main/java/com/softserveinc/ita/homeproject/homeservice/mapperViewToDto/DtoToViewMapper.java [hunk 7]
```diff
@@ -0,0 +1,8 @@
+package com.softserveinc.ita.homeproject.homeservice.mapperViewToDto;
+
+import com.softserveinc.ita.homeproject.homeservice.dto.BaseDto;
+
+public interface DtoToViewMapper<ServiceDto extends BaseDto, ViewDto> {
```
Comment (kh0ma) [inline home-service/src/main/java/com/softserveinc/ita/homeproject/homeservice/mapperViewToDto/DtoToViewMapper.java]: Can we introduce base View in design?

**Review by kh0ma — CHANGES_REQUESTED** [review]

File: home-application/src/main/java/com/softserveinc/ita/homeproject/application/apiservice/NewsApiServiceImpl.java (hunk 1 above)
Comment (alexorlenko) [reply home-application/src/main/java/com/softserveinc/ita/homeproject/application/apiservice/NewsApiServiceImpl.java]: Done

File: home-application/src/main/java/com/softserveinc/ita/homeproject/application/apiservice/NewsApiServiceImpl.java (hunk 2 above)
Comment (alexorlenko) [reply home-application/src/main/java/com/softserveinc/ita/homeproject/application/apiservice/NewsApiServiceImpl.java]: Done

File: home-data/src/main/java/com/softserveinc/ita/homeproject/homedata/entity/BaseEntity.java (hunk 4 above)
Comment (alexorlenko) [reply home-data/src/main/java/com/softserveinc/ita/homeproject/homedata/entity/BaseEntity.java]: Done

File: home-data/src/main/java/com/softserveinc/ita/homeproject/homedata/entity/BaseEntity.java (hunk 3 above)
Comment (alexorlenko) [reply home-data/src/main/java/com/softserveinc/ita/homeproject/homedata/entity/BaseEntity.java:15]: Done

**Review by alexorlenko — COMMENTED** [review]
Yes we will change

File: home-service/src/main/java/com/softserveinc/ita/homeproject/homeservice/dto/NewsDto.java (hunk 6 above)
Comment (alexorlenko) [reply home-service/src/main/java/com/softserveinc/ita/homeproject/homeservice/dto/NewsDto.java]: Done

File: home-service/src/main/java/com/softserveinc/ita/homeproject/homeservice/mapperViewToDto/DtoToViewMapper.java (hunk 7 above)
Comment (alexorlenko) [reply home-service/src/main/java/com/softserveinc/ita/homeproject/homeservice/mapperViewToDto/DtoToViewMapper.java]: Done

File: home-service/src/main/java/com/softserveinc/ita/homeproject/homeservice/dto/BaseDto.java (hunk 5 above)
Comment (alexorlenko) [reply home-service/src/main/java/com/softserveinc/ita/homeproject/homeservice/dto/BaseDto.java:3]: not now

**Review by Romanov-Niko — APPROVED** [review]

**Review by Dandria1802 — APPROVED** [review]

**Review by PoLytvy — APPROVED** [review]

**Review by Abhai2016 — APPROVED** [review]

File: pom.xml [hunk 8]
```diff
@@ -50,8 +52,19 @@
...
             <optional>true</optional>
         </dependency>
+        <dependency>
+            <groupId>org.mapstruct</groupId>
+            <artifactId>mapstruct</artifactId>
+            <version>${mapstruct.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.mapstruct</groupId>
+            <artifactId>mapstruct-processor</artifactId>
```
Comment (kh0ma) [inline pom.xml:65]: This dependency was added several times including distribution management. Please make it once. The same for `mapstruct`.

**Review by kh0ma — APPROVED** [review]
Approve with suggested changes

**Pull request merged**